lib/idn.c

```diff
@@ -50,6 +50,80 @@
 #include "curl_memory.h"
 #include "memdebug.h"
 
+#if defined(__APPLE__)
+/* for macOS and iOS targets */
+
+/* headers included in Xcode 15, so we check for the SDK versions */
+#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 170000
+#define XCODE15 1
+#endif
+
+#if __MAC_OS_X_VERSION_MAX_ALLOWED >= 140000
+#define XCODE15 1
+#endif
+
+#if XCODE15
+#include <unicode/uidna.h>
```

Comment:
Also available in macOS SDK 13:
```
/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/unicode/uidna.h
/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/lib/libicucore.tbd
```

Reply:
Strange, because the CI reported the header missing!?

If the header should exist, then the CI should not fail and the check for SDK versions is not needed.

Reply:
That's odd, but the header and lib does exist on this outdated CommandLineTools install. It also compiles and links and returns a non-NULL address for `uidna_openUTS46(UIDNA_CHECK_BIDI, &err)`.

Do you have a link for the CI failure?

Reply:
This is fixed with __has_include() check. Compiles using the feature if you have the header, but not if it is missing.